packages/vue/src/sdk.ts

```diff
@@ -0,0 +1,416 @@
+    $_sentryPerfHook?: boolean;
+  };
+  $once(hook: string, cb: () => void): void;
+}
+
+/** Optional metadata attached to Sentry Event */
+interface Metadata {
+  [key: string]: any;
+  componentName?: string;
+  propsData?: { [key: string]: any };
+  lifecycleHook?: string;
+}
+
+// https://vuejs.org/v2/api/#Options-Lifecycle-Hooks
+type Hook =
+  | 'activated'
+  | 'beforeCreate'
+  | 'beforeDestroy'
+  | 'beforeMount'
+  | 'beforeUpdate'
+  | 'created'
+  | 'deactivated'
+  | 'destroyed'
+  | 'mounted'
+  | 'updated';
+
+type Operation = 'activate' | 'create' | 'destroy' | 'mount' | 'update';
+
+// Mappings from operation to corresponding lifecycle hook.
+const HOOKS: { [key in Operation]: Hook[] } = {
+  activate: ['activated', 'deactivated'],
+  create: ['beforeCreate', 'created'],
+  destroy: ['beforeDestroy', 'destroyed'],
+  mount: ['beforeMount', 'mounted'],
+  update: ['beforeUpdate', 'updated'],
+};
+
+const COMPONENT_NAME_REGEXP = /(?:^|[-_/])(\w)/g;
+const ROOT_COMPONENT_NAME = 'root';
+const ANONYMOUS_COMPONENT_NAME = 'anonymous component';
+
+/** Vue specific configuration for Tracing Integration  */
+interface TracingOptions {
+  /**
+   * Decides whether to track components by hooking into its lifecycle methods.
+   * Can be either set to `boolean` to enable/disable tracking for all of them.
+   * Or to an array of specific component names (case-sensitive).
+   */
+  trackComponents: boolean | string[];
+  /** How long to wait until the tracked root activity is marked as finished and sent of to Sentry */
+  timeout: number;
+  /**
+   * List of hooks to keep track of during component lifecycle.
+   * Available hooks: 'activate' | 'create' | 'destroy' | 'mount' | 'update'
+   * Based on https://vuejs.org/v2/api/#Options-Lifecycle-Hooks
+   */
+  hooks: Operation[];
+}
+
+/**
+ * Inits the Vue SDK
+ */
+export function init(
+  options: Partial<Omit<VueOptions, 'tracingOptions'> & { tracingOptions: Partial<TracingOptions> }> = {},
+): void {
+  if (options.defaultIntegrations === undefined) {
+    options.defaultIntegrations = defaultIntegrations;
+  }
+  if (options.release === undefined) {
+    const window = getGlobalObject<Window>();
+    // This supports the variable that sentry-webpack-plugin injects
+    if (window.SENTRY_RELEASE && window.SENTRY_RELEASE.id) {
+      options.release = window.SENTRY_RELEASE.id;
+    }
+  }
+
+  const finalOptions = {
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+    Vue: getGlobalObject<any>().Vue as VueInstance,
+    attachProps: true,
+    logErrors: false,
+    tracing: false,
+    ...options,
+    tracingOptions: {
+      hooks: ['activate', 'mount', 'update'],
+      timeout: 2000,
+      trackComponents: false,
+      ...options.tracingOptions,
+    },
+  } as VueOptions;
+
+  initAndBind(BrowserClient, finalOptions);
+  const client = getCurrentHub().getClient();
+  if (client) {
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
```

Comment:
What kind of magic is that? ;o I don't see it being used anywhere.

Reply:
It allows `@ts-ignore` which helps us set `client.__vueHelper`

Reply:
I was actually asking about this `__vueHelper` thing. Why do we need it? I dont see it being used anywhere?